lib/ansible/cli/config.py

```diff
@@ -0,0 +1,180 @@
+            self.parser.set_usage("usage: %prog update [options] [-c ansible.cfg] -s '[section.]setting=value'")
+
+        self.options, self.args = self.parser.parse_args()
+        display.verbosity = self.options.verbosity
+
+    def run(self):
+
+        super(ConfigCLI, self).run()
+
+        if self.options.config_file:
+            self.config_file = unfrackpath(self.options.config_file, follow=False)
+            self.config = ConfigManager(self.config_file)
+        else:
+            self.config = ConfigManager()
+            self.config_file = self.config.data.get_setting('ANSIBLE_CONFIG')
+            try:
+                if not os.path.exists(self.config_file):
+                    raise AnsibleOptionsError("%s does not exist or is not accessible" % (self.config_file))
+                elif not os.path.isfile(self.config_file):
+                    raise AnsibleOptionsError("%s is not a valid file" % (self.config_file))
+
+                os.environ['ANSIBLE_CONFIG'] = self.config_file
+            except:
+                if self.action in ['view']:
+                    raise
+                elif self.action in ['edit', 'update']:
+                    display.warning("File does not exist, used empty file: %s" % self.config_file)
+
+        self.execute()
+
+    def execute_update(self):
+        '''
+        Updates a single setting in the specified ansible.cfg
+        '''
+        raise AnsibleError("Option not implemented yet")
+
+        if self.options.setting is None:
+            raise AnsibleOptionsError("update option requries a setting to update")
+
+        (entry, value) = self.options.setting.split('=')
+        if '.' in entry:
+            (section, option) = entry.split('.')
+        else:
+            section = 'defaults'
+            option = entry
+        subprocess.call([
+            'ansible',
+            '-m','ini_file',
+            'localhost',
+            '-c','local',
+            '-a','"dest=%s section=%s option=%s value=%s backup=yes"' % (self.config_file, section, option, value)
+        ])
+
+    def execute_view(self):
+        '''
+        Displays the current config file
+        '''
+        try:
+            self.pager(open(self.config_file, 'rb').read())
+        except Exception as e:
+            raise AnsibleError("Failed to open config file: %s" % to_native(e))
+
+    def execute_edit(self):
+        '''
+        Opens ansible.cfg in the default EDITOR
+        '''
+        raise AnsibleError("Option not implemented yet")
+        try:
+            editor = shlex.split(os.environ.get('EDITOR','vi'))
+            editor.append(self.config_file)
+            subprocess.call(editor)
+        except Exception as e:
+            raise AnsibleError("Failed to open editor: %s" % to_native(e))
+
+    def execute_list(self):
+        '''
+        list all current configs reading lib/constants.py and shows env and config file setting names
+        '''
```

Comment:
Found that the above use of pager can yield tracebacks so need to fix it here too.